packages/app/components/Today/QueueCreateButton.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { ReactElement, useState } from "react";
+import { useProfile } from "../../hooks/useProfile";
+import { Role } from "@koh/common";
+import { CheckinButton } from "./TACheckinButton";
+import { useRouter } from "next/router";
+import { useCourse } from "../../hooks/useCourse";
+import { useRoleInCourse } from "../../hooks/useRoleInCourse";
+import { Col, Form, Input, message, Modal, Row, Switch } from "antd";
+import TextArea from "antd/lib/input/TextArea";
+import { API } from "@koh/api-client";
+
+export default function TodayPageCreateButton(): ReactElement {
+  const profile = useProfile();
+  const router = useRouter();
+  const [modalVisible, setModalVisible] = useState(false);
+  const { cid } = router.query;
+  const { course } = useCourse(Number(cid));
+  const role = useRoleInCourse(Number(cid));
+  const queueCheckedIn = course?.queues.find((queue) =>
+    queue.staffList.find((staff) => staff.id === profile?.id)
+  );
+  const [form] = Form.useForm();
+
+  async function submitMakeQueue() {
+    const queueRequest = await form.validateFields();
+    try {
+      await API.taStatus.makeQueue(
+        Number(cid),
+        queueRequest.officeHourName,
+        !queueRequest.allowTA,
+        queueRequest.notes
+      );
+
+      message.success(
+        `created a new queue ${queueRequest.officeHourName}. Checking you in...`
+      );
+
+      const redirectID = await API.taStatus.checkIn(
+        Number(cid),
+        queueRequest.officeHourName
+      );
+
+      router.push(
+        "/course/[cid]/queue/[qid]",
+        `/course/${Number(cid)}/queue/${redirectID.id}`
+      );
+    } catch (err) {
+      message.error(err.response?.data?.message);
+    }
+  }
+
+  return (
+    <>
+      {modalVisible && (
+        <Modal
+          title={`Create a new queue`}
+          visible={modalVisible}
+          onCancel={() => {
+            setModalVisible(false);
+          }}
+          okText="Create"
+          onOk={submitMakeQueue}
+        >
+          <Form form={form}>
+            <Row>
```

Comment:
Gonna be *that person* but just a nit: a [Checkbox](https://ant.design/components/checkbox/) would probably look a bit less awkward?